RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -97,13 +97,42 @@ private void ReplaceAssignedByValParameterReferences(IModuleRewriter rewriter, s
             }
         }
 
+
         private void InsertLocalVariableDeclarationAndAssignment(IModuleRewriter rewriter, string localIdentifier)
         {
-            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine
-                + (_target.AsTypeDeclaration is ClassModuleDeclaration ? Tokens.Set + " " : string.Empty)
-                + localIdentifier + " = " + _target.IdentifierName;
+            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine;
+            if (IsBaseTypeContext(_target))
+            {
+                content = content + localIdentifier + " = " + _target.IdentifierName;
+            }
+            else 
+            {
+                //All we can know is that it is not a Base type.  Let VBA determine
+                //the right way to assign the parameter.  The user can simplify it later.
+                string insertIsObjectCheck =
+@"If(IsObject({1})) Then
```

Comment:
This PR broadens the coverage of detection for both ObjectVariableNotSet and the AssignedByValParameter...QuickFix.  The PR is now enhanced to detect and evaluate Variant type assignments.  Todo: When a variant is assigned in a manner like `myVar = Range("A1:C1")` - the need to use Set is not presently determined correctly by evaluating the RHS (the test for this case has been marked 'Ignore').  Added a class to avoid duplicating the 'Set' detection code in two classes.  The new VariableRequiresSetAssignmentEvaluator class is used by both the ObjectVariableNotSet inspection and the AssignedByValParamete...QuickFix.  The class  looks  like a candidate for Constructor Injection to me...but I'll let others decide that.